### PR TITLE
Don't show shield next to deleted messages

### PR DESCRIPTION
**Pull request by SimonBrandner**
Fixes https://github.com/vector-im/element-web/issues/20475
Type: defect
Requires https://github.com/matrix-org/matrix-js-sdk/pull/2143

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Don't show shield next to deleted messages ([\#7671](https://github.com/matrix-org/matrix-react-sdk/pull/7671)). Fixes vector-im/element-web#20475. Contributed by @SimonBrandner.

### CODE REVIEW

File: src/components/views/rooms/EventTile.tsx [hunk 1]
```diff
@@ -1611,8 +1618,8 @@ function isMessageEvent(ev: MatrixEvent): boolean {
 }
 
 export function haveTileForEvent(e: MatrixEvent, showHiddenEvents?: boolean): boolean {
-    // Only messages have a tile (black-rectangle) if redacted
-    if (e.isRedacted() && !isMessageEvent(e)) return false;
+    // Only show "Message deleted" tile for message or encrypted events
+    if (e.isRedacted() && !e.isEncrypted() && !isMessageEvent(e)) return false;
```
Comment (t3chguy) [inline src/components/views/rooms/EventTile.tsx:1622]: I'm not sure I understand the need for this change, why would only encrypted events show that they were redacted?

Comment (SimonBrandner) [reply src/components/views/rooms/EventTile.tsx:1622]: Previously, we showed the `Message removed` tile only for `m.room.message` events. But if an event is encrypted we can't tell if it's an `m.room.message` event, so we always show the tile for encrypted events

Comment (t3chguy) [reply src/components/views/rooms/EventTile.tsx:1622]: Ah fun

**Review by t3chguy — APPROVED** [review]
LGTM - thanks

**Pull request merged**